app/code/Magento/Catalog/Model/Product/Attribute/Backend/TierPrice/SaveHandler.php

```diff
@@ -132,12 +135,12 @@ private function getAdditionalFields(array $objectArray): array
      * Check whether price has percentage value.
      *
      * @param array $priceRow
-     * @return integer|null
+     * @return float|null
      */
-    private function getPercentage(array $priceRow)
+    private function getPercentage(array $priceRow): ?float
     {
         return isset($priceRow['percentage_value']) && is_numeric($priceRow['percentage_value'])
-            ? (int)$priceRow['percentage_value']
+            ? (float)$priceRow['percentage_value']
```

Comment:
There is no need to cast explicitly to `float`, usually `+ 0` or `* 1` is used to convert so that no unneeded int <-> float occur.

Reply:
Shouid I do it
• $priceRow['percentage_value'] + 0
or it
• $priceRow['percentage_value'] * 1
???

Reply:
Yes, of course.